### PR TITLE
[Frontend_v2] Edit Challenge details Feature

### PR DESCRIPTION
**Pull request by Kajol-Kumari**
**Changes introduced through this PR**

 

- [x] `Edit start and end date` feature added to the challenge setting tab.

 

- [x] `Edit Challenge Image` feature added to the challenge setting tab. 

- [x] `Close participation` feature moved to `settings` tab of the challenge.

- [x] `Delete challenge` feature move do `settings` tab of the challenge.

- [x]  `Edit title` feature moved to `settings` tab of the challenge.

GIF:
edit-challenge

### CODE REVIEW

**Comment by lunayach** [comment]
@Kajol-Kumari Thanks a lot for doing a great job in the PR's. Maybe it's a good idea to add labels for all the frontend PRs which are ready for review in this repo. Would just help in filtering them for us.

**Comment by RishabhJain2018** [comment]
Yes, I agree. @Kajol-Kumari Can you please make that change for the dev environment only.

**Comment by Kajol-Kumari** [comment]
Hey @Shekharrajak and @RishabhJain2018 , I will be opening a separate PR for adding logger for dev environment. So, I request you to please review this one.

File: frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts [hunk 1]
```diff
@@ -1,10 +1,13 @@
 import { Component, OnInit } from '@angular/core';
+import {COMMA, ENTER} from '@angular/cdk/keycodes';
+import {MatChipInputEvent} from '@angular/material/chips';
```
Comment (lunayach) [inline frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts:3]: Minor spacing issue

File: frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts [hunk 2]
```diff
@@ -155,4 +174,237 @@ export class ChallengesettingsComponent implements OnInit {
...
+          'is_registration_open': !this.challenge['is_registration_open']
+        });
+        this.apiService.patchUrl(
+          this.endpointsService.editChallengeDetailsURL(this.challenge.creator.id, this.challenge.id),
+          BODY
+        ).subscribe(
+          () => {
+            this.challenge['is_registration_open'] = !this.challenge['is_registration_open'];
+            this.globalService.showToast(
+              'success', 'Participation is ' + participationState.replace('n', 'ne') + 'd successfully', 5
```
Comment (lunayach) [inline frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts:197]: Please create a separate string variable for this. Just looks complicated for no reason.

File: frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts [hunk 3]
```diff
@@ -155,4 +174,237 @@ export class ChallengesettingsComponent implements OnInit {
...
+        ).subscribe(
+        data => {
+          SELF.challenge.title = data.title;
+          SELF.globalService.showToast('success', 'The challenge title is  successfully updated!', 5);
+        },
+        err => {
+          SELF.globalService.handleApiError(err, true);
+          SELF.globalService.showToast('error', err);
+        },
+        () =>  console.log('EDIT-CHALLENGE-TITLE-FINISHED')
```
Comment (lunayach) [inline frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts:239]: Remove this logging statement.

File: frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts [hunk 4]
```diff
@@ -155,4 +174,237 @@ export class ChallengesettingsComponent implements OnInit {
...
+        ).subscribe(
+        data => {
+          SELF.router.navigate([redirectTo]);
+          SELF.globalService.showToast('success', 'The Challenge is successfully deleted!', 5);
+        },
+        err => {
+          SELF.globalService.handleApiError(err, true);
+          SELF.globalService.showToast('error', err);
+        },
+        () => console.log('DELETE-CHALLENGE-FINISHED')
```
Comment (lunayach) [inline frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts:284]: Remove this as well.

File: frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts [hunk 5]
```diff
@@ -155,4 +174,237 @@ export class ChallengesettingsComponent implements OnInit {
...
+          data => {
+            SELF.challenge.start_date = (data.start_date);
+            SELF.challenge.end_date = (data.end_date);
+            SELF.globalService.showToast('success', 'The Challenge start and end date successfully updated!', 5);
+          },
+          err => {
+            SELF.globalService.handleApiError(err, true);
+            SELF.globalService.showToast('error', err);
+          },
+          () => console.log('EDIT-CHALLENGE-START-AND-END-DATE-FINISHED')
```
Comment (lunayach) [inline frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts:332]: This.

File: frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts [hunk 6]
```diff
@@ -155,4 +174,237 @@ export class ChallengesettingsComponent implements OnInit {
...
+          FORM_DATA
+        ).subscribe(
+          data => {
+            SELF.globalService.showToast('success', 'The Challenge image successfully updated!', 5);
+          },
+          err => {
+            SELF.globalService.handleApiError(err, true);
+            SELF.globalService.showToast('error', err);
+          },
+          () => console.log('EDIT-CHALLENGE-IMAGE-FINISHED')
```
Comment (lunayach) [inline frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts:384]: Remove this

**Comment by deshraj** [comment]
@Kajol-Kumari can you please resolve the conflicts for this PR?

**Comment by RishabhJain2018** [comment]
@Kajol-Kumari Can you please resolve the conflicts here?

**Comment by Kajol-Kumari** [comment]
> @Kajol-Kumari Can you please resolve the conflicts here?

Done! @RishabhJain2018

**Pull request merged**